Orange/canvas/__main__.py

```diff
@@ -694,6 +696,17 @@ def show_message(message):
     gc.collect()
 
     del app
+
+    if status == 96:
+        log.info('Restarting via exit code 96.')
+        try:
+            p = psutil.Process(os.getpid())
+            for handler in p.open_files() + p.connections():
+                os.close(handler.fd)
+        except Exception as e:  # pylint: disable=broad-except
+            logging.error(e)
+        return os.execv(sys.executable, ['python', sys.argv[0]])
```

Comment:
I do not think this approach is correct. Granted that this is close to process exit, but threads could (and do) still run and it is unknown how forcibly closing all fds would affect them.

Similarly plain execv would for the non-main threads be equivalent to SIGKILL, possibly leaving inconsistent persistent state.

I think it would be better to simply start the new process (QProcess.startDetached or just subprocess.Popen), and letting this one exit normally.

Reply:
Good point. I was trying to still run as the 'same process' in the eyes of a command line. I guess this isn't really necessary.

Reply:
Well, QProcess.startDetached was introduced in Qt 5.10 (unviable for Anaconda users on 5.9), and subprocessing.Popen's `DETACHED_PROCESS` flag was introduced in python3.7.

I'll have to do some more reading on subprocessing.Popen, I think it should allow the parent process to exit gracefully regardless of the `DETACHED_PROCESS` flag.